openeo/metadata.py

```diff
@@ -199,149 +199,45 @@ def rename_labels(self, target, source) -> Dimension:
         return BandDimension(name=self.name, bands=new_bands)
 
 
-class CollectionMetadata:
+class CubeMetadata:
     """
-    Wrapper for Image Collection metadata.
-
-    Simplifies getting values from deeply nested mappings,
-    allows additional parsing and normalizing compatibility issues.
-
-    Metadata is expected to follow format defined by
-    https://openeo.org/documentation/1.0/developers/api/reference.html#operation/describe-collection
-    (with partial support for older versions)
+    Interface for metadata of a data cube.
 
+    Allows interaction with the cube dimensions and their labels (if available).
     """
 
-    # TODO: "CollectionMetadata" is also used as "cube metadata" where the link to original collection
-    #       might be lost (if any). Better separation between rich EO raster collection metadata and
-    #       essential cube metadata? E.g.: also thing of vector cubes.
-
-    def __init__(self, metadata: dict, dimensions: List[Dimension] = None):
+    def __init__(self, dimensions: List[Dimension] = None):
         # Original collection metadata (actual cube metadata might be altered through processes)
-        self._orig_metadata = metadata
-
-        if dimensions == None:
-            self._dimensions = self._parse_dimensions(self._orig_metadata)
-        else:
-            self._dimensions = dimensions
+        self._dimensions = None
         self._band_dimension = None
         self._temporal_dimension = None
+
+        if dimensions is not None:
+            self.set_dimensions(dimensions=dimensions)
+
+    def set_dimensions(self, dimensions: List[Dimension]):
```

Comment:
I don't think `set_dimensions` should be a public method. 
Ideally, the metadata classes should be immutable from the outside, so they should not have setters

Reply:
I would also just keep this logic under `__init__`, which will make it easier to understand

Reply:
reason I took this out of __init__ is so that the init of CollectionMetadata could reuse this code after parsing dimensions from metadata. 
I added some todo's about type checking e.g. BandsDimension, that would need to be resolved on two places otherwise.